README.md

```diff
@@ -1,486 +1,679 @@
+**Decorators** `@decorator` are *functions* called on class elements or other JavaScript syntax forms during definition, potentially *wrapping* or *replacing* them with a new value returned by the decorator.
 
-ES6 classes were intentionally minimal, and they don't support some common behaviors needed from classes. Some of these use cases are handled by [class fields](https://github.com/tc39/proposal-class-fields) and [private methods](https://github.com/tc39/proposal-private-methods), but others require some kind of programmability or introspection. Decorators make class declarations programmable.
+A decorated class field is treated as wrapping a getter/setter pair for accessing that storage. Decorated storage is useful for observation/tracking, which has been a pain point for the original legacy/experimental decorators combined with [[Define]] semantics for class fields. These semantics are based on Michel Weststrate's ["trapping decorators" proposal](https://github.com/tc39/proposal-decorators/issues/299).
 
-Decorators are very widely used in JavaScript through transpilers today. For example, see the documentation of [core-decorators](https://www.npmjs.com/package/core-decorators), [ember-decorators](https://ember-decorators.github.io/ember-decorators/), [Angular](https://medium.com/@madhavmahesh/list-of-all-decorators-available-in-angular-71bdf4ad6976), [Stencil](https://stenciljs.com/docs/decorators/), and [MobX](https://mobx.js.org/refguide/modifiers.html) decorators.
+Decorators may also annotate a class element with *metadata*. These are simple, unrestricted object properties, which are collected from all decorators which add them, and made available as a set of nested objects in the `[Symbol.metadata]` property.
 
-A few examples of how to implement and use decorators that are a bit more self-contained:
+By making decorators always simply wrap what they are decorating, rather than performing other transformations, this proposal aims to meet the following requirements:
+- The class "shape" is visible without executing the code, making decorators more optimizable for engines.
+- Implementations can work fully on a per-file basis, with no need for cross-file knowledge.
+- No new namespace or type of second-class value is added--decorators are functions.
 
-### `@logged`
+# Examples
 
-The `@logged` decorator logs a console message when a method starts and finishes. Many other popular decorators will also want to wrap a function, e.g., `@deprecated`, `@debounce`, `@memoize`, dependency injection, etc.
+A few examples of how to implement and use decorators in this proposal:
+
+## `@logged`
+
+The `@logged` decorator logs a console message when a method starts and finishes. Many other popular decorators will also want to wrap a function, e.g., `@deprecated`, `@debounce`, `@memoize`, etc.
 
 Usage:
 
 ```mjs
-import { @logged } from "./logged.mjs";
+import { logged } from "./logged.mjs";
 
 class C {
   @logged
-  method(arg) {
+  m(arg) {
```

Comment:
Should this be `method`? It doesn't match the desugared output below.

Reply:
Thanks, fixed